### PR TITLE
WT-9057 Address null address read in compact walk.

### PR DESCRIPTION
**Pull request by jeremythorp**
WT-9057 Address null address read in compact walk.

### CODE REVIEW

File: src/btree/bt_handle.c [hunk 1]
```diff
@@ -768,9 +824,15 @@ int
 __wt_btree_new_leaf_page(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_BTREE *btree;
+    uint8_t previous_state;
 
+    previous_state = ref->state;
```
Comment (agorrod) [inline src/btree/bt_handle.c:829]: I believe it's too late to save the previous state here - it has already been shifted to locked by the caller.

File: src/btree/bt_handle.c [hunk 2]
```diff
@@ -760,6 +760,62 @@ __btree_tree_open_empty(WT_SESSION_IMPL *session, bool creation)
     return (ret);
 }
 
+/*
+ * __wt_btree_new_internal_page --
+ *     Create an empty internal page, as well as a leaf page that is a child of the internal page.
```
Comment (agorrod) [inline src/btree/bt_handle.c:765]: I think this probably should just create a new empty internal page - that page will be traversed by the walk, and traversing into the page will cause the child to be created. I suggest making that change, before digging into the IKEY problem further.

File: src/btree/bt_handle.c (hunk 1 above)
Comment (jeremythorp) [reply src/btree/bt_handle.c]: Good point!

**Comment by agorrod** [comment]
I'm closing this, since it's been moved to WT-9708 - I added a reference to the branch in that ticket, so it should still be discoverable.

**Pull request closed without merging**